src/SFML/System/Win32/SleepImpl.cpp

```diff
@@ -0,0 +1,55 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2022 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/System/Win32/SleepImpl.hpp>
+#include <SFML/System/Win32/WindowsHeader.hpp>
+#include <mmsystem.h>
+
+namespace sf::priv
+{
+////////////////////////////////////////////////////////////
+void sleepImpl(Time time)
+{
+    // Get the minimum supported timer resolution on this system
+    static const UINT periodMin = []
+    {
+        TIMECAPS tc;
+        timeGetDevCaps(&tc, sizeof(TIMECAPS));
+        return tc.wPeriodMin;
+    }();
```

Comment:
I assume this is to reduce the calls to `timeGetDevCaps` right?

Reply:
Yes, this was originally suggested by @reconndev. It's not really about performance, but more about communicating the fact that `wPeriodMin` will never change -- it's hardware-specific.